app/helpers/flash_messages_helper.rb

```diff
@@ -34,23 +36,19 @@ module FlashMessagesHelper
     include FlashMessagesOutputSafetyHelper
   end
 
-  # Renders flash messages
+  # Renders flash messages.
+  #
+  # @return [String] an HTML-safe string.
   def render_flash_messages
-    messages = flash
-      .reject { |k, _| k.start_with? "_" }
-      .reject { |k, _| k.to_s == "op_modal" }
-      .map { |k, v| render_flash_content(k.to_sym, v) }
-
-    safe_join messages, "\n"
+    safe_join build_flash_components.map { it.render_in(self) }, "\n"
   end
 
-  def render_flash_content(key, content)
-    case content
-    when Hash
-      render_flash_message(key, **content)
-    else
-      render_flash_message(key, message: content)
-    end
+  # Renders flash messages wrapped in `<turbo-stream>` tags, suitable for
+  # inclusion inline (e.g. as part of a turbo-frame response).
+  #
+  # @return [String] an HTML-safe string.
+  def render_flash_messages_as_turbo_streams
+    safe_join(build_flash_components.map { it.render_as_turbo_stream(view_context: self, action: :flash) })
```

Comment:
As noted in the PR description, this pattern for displaying flash messages from a Turbo Frame (via `render_flash_messages_as_turbo_streams`) is new. Feedback from @klaustopher @bsatarnejad would be very much appreciated!